### PR TITLE
Add test_single_replica_failed_during_engine_start() skeleton

### PR DESCRIPTION
**Pull request by shuo-wu**
longhorn/longhorn#1253

### CODE REVIEW

File: manager/integration/tests/test_ha.py [hunk 1]
```diff
@@ -836,3 +836,27 @@ def test_rebuild_with_inc_restoration(
...
+    Test if the volume still works fine when there is
+    an invalid replica/backend in the engine starting phase.
+
+    1. Create a pod using Longhorn volume.
+    2. Write some data to the volume then get the md5sum.
+    3. Create a snapshot.
+    4. Repeat step2 and step3 for 3 times then there should be 3 snapshots.
+    5. Randomly pick up a replica and
+       manually messing up the snapshot meta file.
+    6. Delete the pod and wait for the volume detached.
```
Comment (sowmyav27) [inline manager/integration/tests/test_ha.py:853]: Why are we deleting the pod here after corrupting the data? For a replica rebuild?

**Comment by khushboo-rancher** [comment]
Test looks good, just curious if we find a way to figure out the replica is bad based on snapshot data not the meta data https://github.com/longhorn/longhorn/issues/1298, then we will have to add another scenario to mess with snapshot data to make replica go bad.

File: manager/integration/tests/test_ha.py (hunk 1 above)
Comment (shuo-wu) [reply manager/integration/tests/test_ha.py:853]: Deleting the pod will trigger volume detach. The purpose of this step is to detach the volume.

**Comment by shuo-wu** [comment]
> longhorn/longhorn#1298

Unfortunately, Longhorn may not be able to detect if one replica snapshot data is messed up if the replica r/w itself doesn't error out. @yasker Do we need to consider this part? I remembered that we had discusses if we need to add the replica consistency check before attaching and the conclusion was No since the check may take a long time.

**Comment by yasker** [comment]
> > [longhorn/longhorn#1298](https://github.com/longhorn/longhorn/issues/1298)
> 
> Unfortunately, Longhorn may not be able to detect if one replica snapshot data is messed up if the replica r/w itself doesn't error out. @yasker Do we need to consider this part? I remembered that we had discusses if we need to add the replica consistency check before attaching and the conclusion was No since the check may take a long time.

Snapshot data check is out of scope. And currently, we don't have a way to detect the data corruption inside the replica automatically.

**Pull request merged**